### ISSUE TITLE
Decoding of oneof fields only works for the last specified field

### ISSUE THREAD

**Issue by rgilton** [eigenein/protobuf#171]
I am finding that only the last specified field can be successfully decoded for a `oneof` field.

For example, the assertion in this round-trip test fails:

```python
#!/usr/bin/env python3
from pure_protobuf.annotations import Field
from pure_protobuf.message import BaseMessage
from pure_protobuf.one_of import OneOf
from typing_extensions import Annotated
from dataclasses import dataclass
from typing import ClassVar, Optional


@dataclass
class Message(BaseMessage):
    payload: ClassVar[OneOf] = OneOf()
    which_payload = payload.which_one_of_getter()

    foo: Annotated[Optional[int], Field(1, one_of=payload)] = None
    bar: Annotated[Optional[bool], Field(2, one_of=payload)] = None


message = Message()
message.foo = 3

message = Message.loads(message.dumps())
print(message)
assert message.foo == 3
```

The output from the `print` statement is:
```
Message(foo=None, bar=None)
```

If the example is modified so it sets and asserts against the `bar` field, then it works fine.

**Comment by eigenein** [comment]
👋 Thanks for reporting it, I'll have a look!

**Comment by Alesh** [comment]
I has made [pull request](https://github.com/eigenein/protobuf/pull/172), in my case this fix worked. The project's tests were also successful.

**Comment by eigenein** [comment]
The fix is released in `3.1.2`
